File: Searching/Binary_Search/Binary_Search.java
class Binary_Search
{
    // Function for binary search
    public static int BinarySearch(int[] array, int size, int desired)
    {
        int left = 0, right = size - 1, middle;

        while (left <= right)
        {
            middle = left + (right - left) / 2;

            if (array[middle] == desired)
                return middle;
            else if (desired < array[middle])
                right = middle - 1;
            else
                left = middle + 1;
        }

        return -1;
    }

    // Driver Function
    public static void main(String[] args)
    {
        int[] array = {1, 2, 3, 4, 5, 6, 7};

        // Element 4 to be searched
        if (BinarySearch(array, 7, 4) != -1)
            System.out.println("Found");
        else
            System.out.println("Not Found");

        //Element 9 to be searched
        if (BinarySearch(array, 7, 9) != -1)
            System.out.println("Found");
        else
            System.out.println("Not Found");
    }
}

/* Output

Found
Not Found

*/

File: Searching/Binary_Search/Binary_Search.py
# Function for Binary Search
def Binary_Search(list, desired):
    left = 0
    right = len(list) - 1

    while left <= right:
        # Return positon if found
        middle = left + int((right - left) / 2)

        if list[middle] == desired:
            return middle
        elif desired < list[middle]:
            right = middle - 1
        elif desired > list[middle]:
            left = middle + 1

    return -1

list = [1, 2, 3, 4, 5, 6, 7]

# Element to be searched is 4
print ("Found") if Binary_Search(list, 4) != -1 else ("Not Found")

# Element to be searched is 9
print ("Found") if Binary_Search(list, 9) != -1 else ("Not Found")


''' Output

Found
Not Found

'''
